### PR TITLE
Add more diags and fix broken journalctl call

### LINKED ISSUE #738: diags should save results of `ip addr`
As the title states, it would be useful if the `calicoctl diags` command would, in addition to saving `ip route`, also saved `ip addr` output in the diags bundle.

### PR DESCRIPTION
**Pull request by robbrockbank**
Fixes #738

### CODE REVIEW

**Comment by robbrockbank** [comment]
Casey could you review this one?

Primary purpose was to add "ip addr" output to diags.  However, I was also hitting exceptions running the diags command due to the journalctl changes.  I've updated to use the DiagsFileWriter which catches all exceptions and writes a generic log to the file rather than bailing out of the diags script.

**Comment by robbrockbank** [comment]
Fixes #738

File: calicoctl/calico_ctl/diags.py [hunk 1]
```diff
@@ -144,23 +160,22 @@ def save_diags(log_dir, runtime="docker"):
...
+            journalctl = sh.Command._create("journalctl")
+            f.write(journalctl("-u", "calico-node.service", "--no-pager"))
+        except sh.CommandNotFound as e:
+            print "  - Missing command: %s" % e.message
+            f.writelines("Missing command: %s\n" % e.message)
+        except sh.ErrorReturnCode_1 as e:
+            print "  - Error running journalctl."
+            f.writelines("Error running journalctl: %s\n" % e)
+
+    # Ask Felix to dump stats to its log file - ignore errors as the
```
Comment (caseydavenport) [inline calicoctl/calico_ctl/diags.py]: I think this bit about Felix needs to go before the journalctl dump.  When running under rkt, calico/node logs all go to journalctl.

Comment (robbrockbank) [inline calicoctl/calico_ctl/diags.py]: Ah gotcha - I'll beef up the comments then,

**Comment by caseydavenport** [comment]
One comment, once that is done it should be good to go.

**Pull request merged**